### DISCUSSION TITLE
Using custom SSL Certificate or customizing --experimental-https flag with Next Dev Server

### DISCUSSION THREAD

**Discussion by mustafaakin** [vercel/next.js#57916] [Help]
### Summary

I have already my own certificates that I handle wildcard domains. I used `--experimental-https` but it generates its own for localhost. How can I do it?

Normally I use it:

```
mkcert -key-file key.pem -cert-file cert.pem mydomain.com "*.mydomain.com"
mkcert -install
```

I could replace those certificates generated by next with my own, but I'm not sure thats ideal.

Also `npx next dev -h` does not yield any additional information if any further configuration options exits.

Upon search I found those flags:

```
--experimental-https-cert=cert.pem --experimental-https-key=cert.key which works as well. 
```

**Comment by bvoigtlaender** [comment]
I did the mistake to believe that you wouldn't need `--experimental-https` when providing your own key and certificate. That is not the case you need to provide all three arguments.

`next dev --experimental-https --experimental-https-key ./certificates/com.example.key --experimental-https-cert ./certificates/com.example.crt`
